mangdangroboticsclub/minipupper_ros
# Update for mini_pupper_driver

## Proposed change(s)
I have separated the basic control part of the Mini Pupper into the mini_pupper_driver package. This package includes a launch file for the vel_to_servo_control_interface, vel_to_servo_control_interface node, and dependencies such as MangDang, src, and pupper. These changes will make the mini_pupper_driver a modular package for driving Mini Pupper separately, not in mini_pupper_bringup or mini_pupper_control.

### Types of change(s)


## Testing and Verification
To verify these changes, I performed the following tests:

1. Run the bringup launch file on Mini Pupper.
2. Check the state log in the console.
3. Run the Keyboard node.
for more information, check the new README.

## Test Configuration

__Mini Pupper Version__  
Mini Pupper

__Raspberry Pi OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

__PC OS + ROS version__  
Ubuntu 22.04, ROS 2 Humble

## Checklist


## Other comments
1. The new mini_pupper_bringup package depends on the mini_pupper_driver package.
2. To test the mini_pupper_driver package, you should test mini_pupper_bringup together.
3. The dependencies should be reformed in a proper architecture and may be adjusted in future work on redesigning the Mangdang bottom layer, which will replace the StanfordQuadrupedRobot package.
